Promises/Promises/V1/Core/Promise.cs

```diff
@@ -726,16 +726,18 @@ private void AddResolveHandler(Action resolveCallback)
             switch (State)
             {
                 case PromiseState.Failed:
-                    if (resolveCallback != null)
+                    if (resolveCallback == null)
                     {
-                        _pureResolveHandlers.Add(resolveCallback);
+                        throw new ArgumentNullException("resolveCallback");
```

Comment:
Can you change to throw new `ArgumentNullException(nameof(resolveCallback));` instead of string literals? It's cleaner and easier to do future refactors